jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -257,21 +266,19 @@ object XGBoost extends Serializable {
    * @param useExternalMemory indicate whether to use external memory cache, by setting this flag as
    *                          true, the user may save the RAM cost for running XGBoost within Spark
    * @param missing the value represented the missing value in the dataset
-   * @param baseMargin initial prediction for boosting.
    * @throws ml.dmlc.xgboost4j.java.XGBoostError when the model training is failed
    * @return XGBoostModel when successful training
    */
   @throws(classOf[XGBoostError])
   def trainWithRDD(
-      trainingData: RDD[MLLabeledPoint],
+      trainingData: RDD[XGBLabeledPoint],
```

Comment:
I looked at this issue for the second time and think about the solution

1. since XGBoost has claimed to only support Spark 2 since a long time ago, it's time to make it move forward further, let's only add feature to DataFrame-based APIs (including this one). In this way, we can also resolve the issue about how to persist data in data engineering phase (we only need to ensure that we operate on DataFrame level)

2. keep this signature change https://github.com/dmlc/xgboost/pull/2532/files#diff-095250777d030169ae83818b37ab44b1R214, but make it as private

3. revert this one, https://github.com/dmlc/xgboost/pull/2532/files#diff-095250777d030169ae83818b37ab44b1R271, i.e. we will not support baseMargin in RDD,